stl/inc/ranges

```diff
@@ -2462,6 +2462,352 @@ namespace ranges {
+                        ++_Outer;
+                        _Satisfy();
+                    }
+                    return *this;
+                } else {
+                    if (++_Inner == _RANGES end(_Parent->_Inner)) {
+                        ++_Outer;
+                        _Satisfy();
+                    }
+                    return *this;
+                }
+            }
+
+            constexpr void operator++(int) {
+                ++*this;
+            }
+
+            // clang-format off
+            constexpr decltype(auto) operator++(int)
+                requires _Deref_is_ref && forward_range<_Base>&& forward_range<_SubRng> {
+                // clang-format on
+                auto _Tmp = *this;
+                ++*this;
+                return _Tmp;
+            }
+
+            // clang-format off
+            constexpr _Iterator& operator--()
+                requires _Deref_is_ref && bidirectional_range<_Base> && bidirectional_range<_SubRng>
+                      && common_range<_SubRng> {
+                // clang-format on
+                if (_Outer == _RANGES end(_Parent->_Range)) {
+                    _Inner = _RANGES end(*--_Outer);
+                }
+                while (_Inner == _RANGES begin(*_Outer)) {
+                    _Inner = _RANGES end(*--_Outer);
+                }
+                --_Inner;
+                return *this;
+            }
+
+            // clang-format off
+            constexpr _Iterator& operator--(int)
+                requires _Deref_is_ref && bidirectional_range<_Base> && bidirectional_range<_SubRng>
+                      && common_range<_SubRng> {
+                // clang-format on
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            // clang-format off
+            _NODISCARD friend constexpr bool operator==(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Outer == _Right._Outer && _Left._Inner == _Right._Inner)) /* strengthened */
+                requires _Deref_is_ref && equality_comparable<_OuterIter> && equality_comparable<_InnerIter> {
+                // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Left._Same_range(_Right);
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _Left._Outer == _Right._Outer && _Left._Inner == _Right._Inner;
+            }
+
+            _NODISCARD friend constexpr decltype(auto) iter_move(const _Iterator& _It) noexcept(
+                noexcept(_RANGES iter_move(_It._Inner))) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _It._Check_dereference();
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _RANGES iter_move(_It._Inner);
+            }
+
+            friend constexpr void iter_swap(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(ranges::iter_swap(_Left._Inner, _Right._Inner))) requires indirectly_swappable<_InnerIter> {

```

Comment:
```suggestion
                noexcept(_RANGES iter_swap(_Left._Inner, _Right._Inner))) requires indirectly_swappable<_InnerIter> {
```

Reply:
I'll correct the source in `transform_view` as well.